### PR TITLE
Feature/faster parsing

### PR DESCRIPTION
**Pull request by chrisputnam9**
Here is the faster parsing feature.

This is a branch off of test/large-json-parse-slow so those tests are included

See #24

### CODE REVIEW

File: src/HJSON/HJSONParser.php [hunk 1]
```diff
@@ -5,7 +5,8 @@
 class HJSONParser
 {
 
-    private $text;
+    private $text_array;
```
Comment (peter279k) [inline src/HJSON/HJSONParser.php:8]: Wrong coding style. It should be `$textArray`.

File: src/HJSON/HJSONParser.php [hunk 2]
```diff
@@ -312,25 +315,37 @@ private function error($m)
...
 
+        // Start with where we're at now, count back to most recent line break
+        // - to determine "column" of error hit
         $i = $this->at;
         while ($i > 0) {
-            $ch = mb_substr(mb_strcut($this->text, $i - 1), 0, 1);
-            $i -= strlen($ch);
 
+            // Mimic old behavior with mb_substr
+            if ($i >= $this->text_length_chars)
```
Comment (peter279k) [inline src/HJSON/HJSONParser.php:324]: Wrong coding style. It should be `if ($i >= $this->text_length_chars) {`.

File: src/HJSON/HJSONParser.php [hunk 3]
```diff
@@ -343,19 +358,25 @@ private function next($c = false)
...
-        // range check is not required
-        if ($offs >= 0) {
-            return mb_substr(mb_strcut($this->text, $this->at), $offs, 1);
-        } else {
-            return mb_substr(mb_strcut($this->text, 0, $this->at), $offs, 1);
-        }
+        $index = $this->at + $offs;
+
+        // Mimic old behavior with mb_substr
+        if ($index < 0) $index = 0;
```
Comment (peter279k) [inline src/HJSON/HJSONParser.php:376]: Wrong coding style. It should be:

```PHP
 if ($index < 0) {
    $index = 0;
}
```

**Review by trobro — APPROVED** [review]
Nice improvement! Sorry for not reviewing sooner.

**Pull request merged**